crates/sui-replay/src/lib.rs

```diff
@@ -293,6 +311,39 @@ pub async fn execute_replay_command(
             // TODO: clean this up
             Some((0u64, 0u64))
         }
+        ReplayToolCommand::BatchReplayFromSandbox { path, num_tasks } => {
+            let files: Vec<_> = std::fs::read_dir(path)?
+                .filter_map(|entry| {
+                    let path = entry.ok()?.path();
+                    if path.is_file() {
+                        path.to_str().map(|p| p.to_owned())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            info!("Replaying {} files", files.len());
+            let chunks = files.chunks(max(files.len() / num_tasks, 1));
+            let tasks = chunks.into_iter().map(|chunk| async move {
+                for file in chunk {
+                    info!("Replaying from state dump file {}", file);
+                    let contents = std::fs::read_to_string(file).unwrap();
+                    let sandbox_state: ExecutionSandboxState =
+                        serde_json::from_str(&contents).unwrap();
+                    let sandbox_state = LocalExec::certificate_execute_with_sandbox_state(
+                        &sandbox_state,
+                        &sandbox_state.pre_exec_diag,
+                    )
+                    .await
+                    .unwrap();
+                    sandbox_state.check_effects().unwrap();
+                }
+            });
+            futures::future::join_all(tasks).await;
+
+            // TODO: WTF is this
```

Comment:
From the call-site in the handling of `ReplayEpoch`, this seems to be the number of successfully executed transactions and the number of total transactions.